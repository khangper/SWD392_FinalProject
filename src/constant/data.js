export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
