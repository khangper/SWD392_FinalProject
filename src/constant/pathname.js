export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
