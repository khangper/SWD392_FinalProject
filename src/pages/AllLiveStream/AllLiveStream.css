/* AllLiveStream.css */

.wrapper {
    padding: 20px;
}

.sa4d25 {
    margin: 20px 0;
}

.container-fluid {
    padding: 0 15px;
}

.grid-container {
    display: grid;
    grid-template-columns: 1fr 300px;
    gap: 20px;
}

.main-content {
    grid-column: 1 / 2;
}

.side-content {
    grid-column: 2 / 3;
}

.la5lo1 {
    display: grid;
}

.grid-row {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
    gap: 20px;
}

.grid-item {
    background: #fff;
    border-radius: 8px;
    overflow: hidden;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
}


.section3125 {
    float: left;
    width: 100%;
}

.item_title {
    font-size: 18px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #333;
    text-align: left;
    float: left;
    margin-bottom: 20px;
}

.la5lo1 {
    float: left;
    width: 100%;
}

.stream_1 {
    width: 100%;
    border-radius: 3px;
    padding: 15px;
    background-position: center;
    background-repeat: no-repeat;
    background: rgba(51, 1, 51, 0.1);
}

.mb-40 {
    margin-bottom: 40px;
}

.stream_bg {
    text-align: center;
    display: inline-block;
    width: 100%;
    padding: 15px 10px;
    border-radius: 3px;
    background: rgba(255, 255, 255, 0.06);
    backdrop-filter: blur(30px);
    box-shadow: 0 6px 25px rgba(0, 0, 0, 0.13);
}

.stream_bg img {
    width: 60px !important;
    height: 60px !important;
    border-radius: 100%;
    display: inline-block !important;
    border: 1px solid #fff;
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.1);
}

.stream_bg h4 {
    margin-top: 10px;
    margin-bottom: 5px;
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    color: #333;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
}

.stream_bg p {
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    color: #686f7a;
    display: inline-block;
    text-align: center;
    position: relative;
}

.stream_bg p span {
    width: 6px;
    height: 6px;
    background: #ed2a26;
    position: absolute;
    border-radius: 100%;
    top: 2px;
    right: -9px;
}

.la5lo1 .owl-theme .owl-nav .disabled {
    opacity: .4 !important;
    cursor: default;
}

.live_stream .owl-nav .owl-prev {
    left: 0px;
    font-size: 20px !important;
    position: absolute;
    top: 40%;
    text-align: center;
    height: 30px !important;
    width: 30px !important;
    padding: 4px 0 !important;
    color: #333 !important;
    background: #fff !important;
    border-radius: 3px !important;
    border: 1px solid #e5e5e5 !important;
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.07);
}

.live_stream .owl-nav .owl-next {
    font-size: 20px !important;
    position: absolute;
    right: 0px;
    top: 40%;
    padding: 4px 0 !important;
    text-align: center;
    height: 30px !important;
    width: 30px !important;
    color: #333 !important;
    background: #fff !important;
    border-radius: 3px !important;
    border: 1px solid #e5e5e5 !important;
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.07);
}

.live_stream .owl-nav .owl-prev:hover {
    color: #fff !important;
    opacity: 1;
    background: #ed2a26 !important;
    border: 1px solid #ed2a26 !important;
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.07);
}

.live_stream .owl-nav .owl-next:hover {
    color: #fff !important;
    opacity: 1;
    border: 1px solid #ed2a26 !important;
    background: #ed2a26 !important;
    box-shadow: 0px 2px 2px 0px rgba(0, 0, 0, 0.07);
}

.main-loader {
    float: left;
    width: 100%;
    margin-bottom: 20px;
}

.mt-20 {
    margin-top: 20px;
}


.spinner {
    margin: 0 auto 20px;
    /* Dịch chuyển spinner xuống dưới và canh giữa */
    width: 50px;
    text-align: center;
    height: 20px;
    line-height: 20px;
    border-radius: 100px;
    background-color: #fff;
    border: 1px solid #efefef;
    cursor: pointer;
    position: relative;
    /* Thêm vị trí tương đối để sử dụng cho các vị trí tuyệt đối bên trong */
}

.spinner>div {
    width: 10px;
    height: 10px;
    border-radius: 100%;
    display: inline-block;
    background-color: #333;
    -webkit-animation: sk-bouncedelay 1.4s infinite ease-in-out;
    /* Thêm tiền tố -webkit- cho hỗ trợ Safari và Chrome */
    animation: sk-bouncedelay 1.4s infinite ease-in-out;
    position: absolute;
    /* Để các quả bóng di chuyển theo vị trí tuyệt đối trong .spinner */
}

.spinner .bounce1 {
    left: 0;
    /* Đặt vị trí ban đầu của bounce1 */
    -webkit-animation-delay: -0.32s;
    /* Độ trễ cho animation */
    animation-delay: -0.32s;
}

.spinner .bounce2 {
    left: 50%;
    /* Đặt vị trí ban đầu của bounce2 */
    transform: translateX(-50%);
    /* Dịch chuyển bounce2 về giữa */
    -webkit-animation-delay: -0.16s;
    animation-delay: -0.16s;
}

.spinner .bounce3 {
    right: 0;
    /* Đặt vị trí ban đầu của bounce3 */
    -webkit-animation-delay: -0.12s;
    animation-delay: -0.12s;
}

@keyframes sk-bouncedelay {

    0%,
    80%,
    100% {
        transform: scale(0);
    }

    40% {
        transform: scale(1);
    }
}

.fcrse_3 {
    background: #fff;
    width: 100%;
    float: left;
    padding: 0;
    border-radius: 3px;
    margin-bottom: 30px;
    border: 1px solid #efefef;
    transition: all .2s ease-in-out;
}

.fcrse_3 h4 {
    font-size: 16px;
    color: #333;
    float: left;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    margin-bottom: 0;
}

.cater_ttle {
    float: left;
    width: 100%;
    padding: 15px 20px;
    border-bottom: 1px solid #efefef;
}

.live_text {
    float: left;
    width: 100%;
    text-align: center;
    padding: 20px;
}

.live_icon {
    float: left;
    width: 100%;
    text-align: center;
    margin-bottom: 20px;
}

.live_icon i {
    display: inline-block;
    width: 70px;
    height: 70px;
    line-height: 43px;
    font-size: 30px;
    text-align: center;
    padding: 14px 0;
    background: #ffecec;
    border-radius: 100%;
    border-bottom: 1px solid #efefef;
}

.live_icon img {
    width: 15%;
}

.live-content {
    float: left;
    width: 100%;
    padding: 0;
    text-align: center;
}

.live-content p {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    line-height: 26px;
    margin-bottom: 20px;
    color: #686f7a;
}

.live_link {
    height: 40px;
    background: #ed2a26;
    border: 0;
    border-radius: 3px;
    padding: 10px 20px;
    color: #fff;
}

.live_link:hover {
    color: #fff;
    background: #333;
}

.livinfo {
    font-size: 12px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    margin-bottom: 0;
    margin-top: 15px;
    color: #686f7a;
    display: block;
}

/*Loading*/

/* .loading-live{
    align-items: center;
    gap: 20px;
    position: absolute;
    top:50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

.loadingLive{
    width: 60px;
    height: 60px;
    border-radius: 50%;
    background-color: #686f7a;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px ;
    animation: animation 1s ease-in-out infinite ;
}

.set_1{
    animation-delay:0s ;
}

.set_2{
    animation-delay:0.25s ;
}

.set_3{
    animation-delay:0.3s ;
}


@keyframes animation {
   0% {
    transform: scale(0.5);
   }
   50% {
    transform: scale(1);
   }
   100% {
    transform: scale(0.5);
   }
} */