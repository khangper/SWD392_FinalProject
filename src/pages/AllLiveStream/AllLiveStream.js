import React from 'react';
import './AllLiveStream.css';
import Kayak from '..//../assets/kayak.png';
import { Link } from 'react-router-dom';
export default function AllLiveStream() {
  return (
    <div className="wrapper">
      <div className="sa4d25">
        <div className="container-fluid">
          <div className="grid-container">
            <div className="main-content">
              <div className="section3125">
                <h4 className="item_title">All Live Streams</h4>
                <div className="la5lo1">
                  <div className="grid-row">
                    {streamData.map((stream, index) => (
                      <div className="grid-item" key={index}>
                        <div className="stream_1 mb-4">
                          <a href="/live-output" className="stream_bg">
                            <img src={stream.img} alt={stream.name} />
                            <h4>{stream.name}</h4>
                            <p>live<span></span></p>
                          </a>
                        </div>
                      </div>
                    ))}
                  </div>
                  <div className="main-loader mt-20">
                    <div className="spinner">
                      <div className="bounce1"></div>
                      <div className="bounce2"></div>
                      <div className="bounce3"></div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div className="side-content">
              <div className="right_side">
                <div className="fcrse_3">
                  <div className="cater_ttle">
                    <h4>Live Streaming</h4>
                  </div>
                  <div className="live_text">
                    <div className="live_icon">
                    <img src={Kayak} alt="Live Icon" />
                    </div>
                    <div className="live-content">
                      <p>Set up your channel and stream live to your students</p>
                      <Link to="/add_livestream">
                      <button className="live_link" >Get Started</button>
                      </Link>
                      
                      <span className="livinfo">Info: This feature is only for 'Instructors'.</span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

const streamData = [
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-1.jpg", name: "John Doe" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-2.jpg", name: "Jassica" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-9.jpg", name: "Edututs+" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-3.jpg", name: "Joginder Singh" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-4.jpg", name: "Zoena" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-5.jpg", name: "Albert Dua" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-6.jpg", name: "Ridhima" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-7.jpg", name: "Amritpal" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-8.jpg", name: "Jimmy" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-11.jpg", name: "Quinton Batchelor" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-12.jpg", name: "Eli Natoli" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-13.jpg", name: "Jaysen Batchelor" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-14.jpg", name: "Farhat Amin" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-15.jpg", name: "Kyle Pew" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-16.jpg", name: "Kerstin Cable" },
  { img: "https://gambolthemes.net/html-items/cursus-new-demo/images/left-imgs/img-17.jpg", name: "Jose Portilla" },
];
