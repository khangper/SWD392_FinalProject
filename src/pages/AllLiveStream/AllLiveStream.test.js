import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import { screen, fireEvent } from "@testing-library/react";
import AllLiveStream from "./AllLiveStream";

describe("AllLiveStream", () => {
  it("renders the AllLiveStream component", () => {
    const { container } = renderWithProviders(<AllLiveStream />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
