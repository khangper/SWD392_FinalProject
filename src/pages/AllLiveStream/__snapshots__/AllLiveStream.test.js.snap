// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllLiveStream renders the AllLiveStream component 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="sa4d25"
    >
      <div
        class="ALL-Live-container-fluid"
      >
        <div
          class="ALL-Live-grid-container"
        >
          <div
            class="main-content"
          >
            <div
              class="section3125"
            >
              <h4
                class="item_title"
              >
                All Live Streams
              </h4>
              <div
                class="la5lo1"
              >
                <div
                  class="AllLive-grid-row"
                />
              </div>
            </div>
          </div>
          <div
            class="side-content"
          >
            <div
              class="right_side"
            >
              <div
                class="fcrse_3"
              >
                <div
                  class="cater_ttle"
                >
                  <h4>
                    Live Streaming
                  </h4>
                </div>
                <div
                  class="live_text"
                >
                  <div
                    class="live_icon"
                  >
                    <img
                      alt="Live Icon"
                      src="kayak.png"
                    />
                  </div>
                  <div
                    class="live-content"
                  >
                    <p>
                      Set up your channel and stream live to your students
                    </p>
                    <a
                      href="/add_livestream"
                    >
                      <button
                        class="live_link"
                      >
                        Get Started
                      </button>
                    </a>
                    <span
                      class="livinfo"
                    >
                      Info: This feature is only for 'Instructors'.
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
